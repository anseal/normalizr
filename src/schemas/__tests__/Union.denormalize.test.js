// eslint-env jest
import { denormalize, schema } from '../../'

describe(`${schema.Union.name} denormalization`, () => {
	const user = new schema.Entity('users')
	const group = new schema.Entity('groups')
	const entities = {
		users: {
			1: { id: 1, username: 'Janey', type: 'users' },
		},
		groups: {
			2: { id: 2, groupname: 'People', type: 'groups' },
		},
	}

	test('denormalizes an object using string schemaAttribute', () => {
		expect(true).toBe(true)
		const union = new schema.Union(
			{
				users: user,
				groups: group,
			},
			'type'
		)

		expect(denormalize({ id: 1, schema: 'users' }, union, entities)).toMatchSnapshot()
		expect(denormalize({ id: 2, schema: 'groups' }, union, entities)).toMatchSnapshot()
	})

	test('denormalizes an array of multiple entities using a function to infer the schemaAttribute', () => {
		const union = new schema.Union(
			{
				users: user,
				groups: group,
			},
			(input) => {
				return input.username ? 'users' : 'groups'
			}
		)

		expect(denormalize({ id: 1, schema: 'users' }, union, entities)).toMatchSnapshot()
		expect(denormalize({ id: 2, schema: 'groups' }, union, entities)).toMatchSnapshot()
	})

	test('returns the original value no schema is given', () => {
		const union = new schema.Union(
			{
				users: user,
				groups: group,
			},
			(input) => {
				return input.username ? 'users' : 'groups'
			}
		)

		expect(denormalize({ id: 1 }, union, entities)).toMatchSnapshot()
	})
})
