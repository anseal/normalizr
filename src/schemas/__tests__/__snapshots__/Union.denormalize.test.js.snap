// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnionSchema denormalization denormalizes an array of multiple entities using a function to infer the schemaAttribute 1`] = `
Object {
  "id": 1,
  "type": "users",
  "username": "Janey",
}
`;

exports[`UnionSchema denormalization denormalizes an array of multiple entities using a function to infer the schemaAttribute 2`] = `
Object {
  "groupname": "People",
  "id": 2,
  "type": "groups",
}
`;

exports[`UnionSchema denormalization denormalizes an object using string schemaAttribute 1`] = `
Object {
  "id": 1,
  "type": "users",
  "username": "Janey",
}
`;

exports[`UnionSchema denormalization denormalizes an object using string schemaAttribute 2`] = `
Object {
  "groupname": "People",
  "id": 2,
  "type": "groups",
}
`;

exports[`UnionSchema denormalization returns the original value no schema is given 1`] = `
Object {
  "id": 1,
}
`;
